File: S/551-student-attendance-record-i/main.go
package main

import "fmt"

func checkRecord(s string) bool {
	A := 0
	L := 0
	isAward := true
	for i := 0; i < len(s); i++ {
		if s[i] == 'A' {
			A++
			if A > 1 {
				isAward = false
				break
			}
		} else if s[i] == 'L' {
			L++
			if L == 3 {
				if s[i-1] == 'L' && s[i-2] == 'L' {
					isAward = false
					break
				} else {
					if s[i-1] == 'L' {
						L = 2
					} else {
						L = 1
					}
				}
			}
		}
	}
	return isAward
}

func main() {
	fmt.Println("a")
}

File: S/551-student-attendance-record-i/main.rs
struct Solution {}

impl Solution {
  pub fn check_record(s: String) -> bool {
    let r = s
      .as_bytes()
      .to_vec()
      .iter()
      .fold((0, false, 0), |(a, l, c), v| match v {
        b'A' => (a + 1, l, 0),
        b'L' => (a, l || c >= 2, c + 1),
        _ => (a, l, 0),
      });
    r.0 < 2 && !r.1
  }
}
